### ISSUE TITLE
Read name

### ISSUE THREAD

**Issue by nikola-tesic** [miRTop/incubator#24]
Hi miRTop guys,

I have a question - what should be the contents of 'Read'? Also, I was reading up on GFF3 documentation [here](http://gmod.org/wiki/GFF3), and it seems that all 'additional' attributes should be lowercase. Does that mean that 'Read' should be lowercase as well?

Best,
Nikola

**Comment by lpantano** [comment]
Hi Nikola,

thanks a lot for this!

Actually you are right, what brings me to think about following the same rule for the miRNA format. Since this is an adaptation from the GFF3, seems right to keep upper case rule for the attributes we define here and let plp add others attributes that should start with lower case. 

I’ll add this as a feature to add to the validator and add the explanation to the documentation.

Thanks a lot for chiming in here! This actually is a very good point and we need to be clearer!

cheers

**Comment by nikola-tesic** [comment]
No problem, it's nice to have a unified format. Also, what should be the contents of 'Read'? Read name from FASTQ file or some generated read name?

**Comment by lpantano** [comment]
Thanks for the question!

It should be the read name that normally is used when mapping to miRNAs.

It could be the name of the fastq file, but normally the tools de-duplicate reads, so same reads from the fastq file are represented more than once are reduced to one sequence with a name that contains the counts. For instance,

@name2_in_fastq
AAAAAAAAAAAAAAAAA
@name1_in_fastq
AAAAAAAAAAAAAAAAA

It will be:

@name_x2
AAAAAAAAAAAAAAAAA

Bottom line is that whatever was used to map to miRNA/precursors…is the name used. This field is not mandatory, but it is useful for debugging proposes.

Cheers

**Comment by nikola-tesic** [comment]
Thank you! So, in your example, would it be OK to have 'Read=name2_in_fastq,name1_in_fastq'? Also, which attribute fields are mandatory and which are optional?

**Comment by lpantano** [comment]
Hi,

That will make a big Read name, since some sequences are million of times repeated.

So, I’d suggest add only the first name in the case you use from the raw fastq files and there is no collapsing previous GFF3 generation.

Thanks!

PS: https://github.com/miRTop/mirGFF3/blob/master/definition.md#columns contains the optionals versus required fields (see O and R before each field)
